base/acme/src/main/java/org/dogtagpki/acme/ACMEPolicy.java

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme;
+
+/**
+ * This class includes mechanisms to enforce various policy and security
+ * restrictions explicitly or implicitly enumerated by ACME.
+ */
+public class ACMEPolicy {
+
+    /**
+     * Validate DNS record from an ACME Order Identifier for wildcard
+     * compliance.
+     *
+     * Throws an exception on malformed wildcards, but ignores non-wildcard
+     * identifiers.
+     */
+    public static void ValidateWildcard(String record) throws Exception {
```

Comment:
The code looks good, but do we want to make the policy customizable?
In that case this method shouldn't be `static` so it can be overridden by a subclass.
I think the `ACMEEngine` should keep an instance of `ACMEPolicy`, then the REST services can get it from `ACMEEngine` if needed.
Also, generally Java methods should start with a lowercase.

Reply:
Yeah I wasn't sure if we wanted a separate `ACMEPolicy` class or if we wanted to add it to `ACMEEngine` directly. Thoughts?

Reply:
I think either way is fine at this point. I don't know yet what else we're going to put in the policy class, and I don't know whether people would prefer to write a custom policy class, or just customize the policy configuration file. We can always refactor the code in the future.

Reply:
How about we also make this return boolean? Then our check in `createNewOrder` can be more robust.

 - true if DNS record is a wildcard of any sort
 - false if DNS record is not a wildcard of any sort
 - Exception if denied by policy / invalid wildcard

Reply:
Something like this?
```
if (engine.isWildcard(identifier)) {
    ...
} else {
    ...
}
```
It probably will make it less apparent that we're applying some kind of policy, but technically it should work.

Reply:
Yeah, see the new commit.